gamemode/mapfixes/triggers.lua

```diff
@@ -17,28 +17,107 @@
 ]]--
 DEFINE_BASECLASS "gamemode_base"
 
+-- Interval between the -200 and 100 damage caused by trigger_hurt that informs us a player is at the end
+local END_CHECK_INTERVAL = 5
+
 local function replaceTrigger(ent)
-    if not ent.kvs then
-        ErrorNoHalt(string.format("Entity %s doesn't have its .kvs?", ent))
-        return
-    end
+  if not ent.kvs then
+    ErrorNoHalt(string.format("Entity %s doesn't have its .kvs?", ent))
+    return
+  end
 
-    -- Take the ent out of the game but don't remove it (Just in case [?])
-    ent:Fire("Disable")
+  -- Take the ent out of the game but don't remove it (Just in case [?])
+  ent:Fire("Disable")
 
-    local replacement = ents.Create('trigger_multiple')
-    for _, kv in ipairs(ent.kvs) do
-        if kv[1] ~= "classname" then
-            replacement:SetKeyValue(kv[1], kv[2])
-        end
+  local replacement = ents.Create('trigger_multiple')
+  for _, kv in ipairs(ent.kvs) do
+    if kv[1] ~= "classname" then
+      replacement:SetKeyValue(kv[1], kv[2])
     end
-    replacement:SetKeyValue("wait", 0)
-    replacement:Spawn()
-    replacement:Activate()
+  end
+  replacement:SetKeyValue("wait", 0)
+  replacement:Spawn()
+  replacement:Activate()
 end
 
 function GM:ReplaceTriggerOnces()
-    for _, ent in pairs(ents.FindByClass("trigger_once")) do
-        replaceTrigger(ent)
+  for _, ent in pairs(ents.FindByClass("trigger_once")) do
+    replaceTrigger(ent)
+  end
+end
+
+--- from: https://wiki.facepunch.com/gmod/ENTITY:TriggerOutput
+function GM:SetupTeleportHook()
+	MapLua = IsValid(MapLua) and MapLua or ents.Create("lua_run")
+	MapLua:SetName("triggerhook")
+	MapLua:Spawn()
+
+	for _, teleport in pairs(ents.FindByClass("trigger_teleport"))do
+		teleport:Fire("AddOutput", "OnStartTouch triggerhook:RunPassedCode:hook.Run('RawOnTeleport'):0:-1")
```

Comment:
Any particular reason to write the code each time rather than setting the `Code` keyvalue?

Reply:
Also why do you use an intermediary hook?

Reply:
I don't understand your first comment, what do you mean with writing the code each time and which Code keyvalue?

Reply:
The intermediary hook is so the "higher level" hook doesn't have to access globals (ACTIVATOR, CALLER). Basically so we can use the hook in a more conventional gmod way.

Reply:
eg
```lua
	MapLua = IsValid(MapLua) and MapLua or ents.Create("lua_run")
	MapLua:SetName("triggerhook")
	MapLua:SetKeyValue(
		"Code", "gamemode.Call('OnPlayerTeleport', ACTIVATOR, CALLER)"
	)
	MapLua:Spawn()

	for _, teleport in pairs(ents.FindByClass("trigger_teleport")) do
		teleport:Fire("AddOutput", "OnStartTouch triggerhook:RunCode::0:-1")
	end
```

Reply:
Oh that's cool, didn't know that was possible. I got that code from here: https://wiki.facepunch.com/gmod/ENTITY:TriggerOutput

Reply:
Sadly the state of gmod and the wiki is you generally shouldn't trust any example there without going and checking out the source first 🙈